MSVC compiler complains about different calling convention between interface and impl when building Win32 binaries

```
\thirdparty\slang-fork\source\compiler-core\slang-artifact-associated.h(143,52): error C2695: “Slang::IArtifac
tPostEmitMetadata::getDebugBuildIdentifier”: overriding virtual function differs from “slang::IMetadata::getDebugBuildIdentifier”
\thirdparty\slang-fork\source\compiler-core\slang-artifact-impl.h(100,52): error C2695: “Slang::Artifact::getM
etadata”: overriding virtual function differs from “slang::IMetadata::getDebugBuildIdentifier”
\thirdparty\slang-fork\source\compiler-core\slang-artifact-impl.h(99,5): error C2695: “Slang::Artifact::getIte
mData”: overriding virtual function differs from......
```


The impl: SLANG_MCALL (stdcall)
The interface: default (thiscall)

This is what I did to fix the compilation error on my end:
<img width="2495" height="624" alt="Image" src="https://github.com/user-attachments/assets/c807428b-fee0-4224-ae1e-5b408fb445d5" />

--- a/include/slang.h
+++ b/include/slang.h
@@ -4201,7 +4201,7 @@ struct IMetadata : public ISlangCastable
     /*
     Returns the debug build identifier for a base and debug spirv pair.
     */
-    virtual const char* getDebugBuildIdentifier() = 0;
+    virtual const char* SLANG_MCALL getDebugBuildIdentifier() = 0;
 };
     #define SLANG_UUID_IMetadata IMetadata::getTypeGuid()
 
@@ -4217,9 +4217,9 @@ struct ICompileResult : public ISlangCastable
         0x41e5,
         {0x9f, 0x12, 0x4b, 0xad, 0x4d, 0x9e, 0xaa, 0xe4})
 
-    virtual uint32_t getItemCount() = 0;
-    virtual SlangResult getItemData(uint32_t index, IBlob** outblob) = 0;
-    virtual SlangResult getMetadata(IMetadata** outMetadata) = 0;
+    virtual uint32_t SLANG_MCALL getItemCount() = 0;
+    virtual SlangResult SLANG_MCALL getItemData(uint32_t index, IBlob** outblob) = 0;
+    virtual SlangResult SLANG_MCALL getMetadata(IMetadata** outMetadata) = 0;
 };
     #define SLANG_UUID_ICompileResult ICompileResult::getTypeGuid()
 

